vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -167,4 +174,21 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
         this.context.eventStream.post(new DotnetAcquisitionDeletion(folderPath));
         rimraf.sync(folderPath);
     }
+
+    private async managePreinstalledVersion(dotnetInstallDir: string, installedVersions: string[]): Promise<string[]> {
+        try {
+            // Determine installed version(s)
+            const versions = fs.readdirSync(path.join(dotnetInstallDir, 'sdk'));
+
+            // Update extension state
+            for (const version of versions) {
```

Comment:
Totally curious, does anything catastrophic happen if there's an extraneous folder in that dir that we're unaware of?

Reply:
Theoretically no, we will add it to our installed versions even if it's an invalid version (or not a version at all) but then it would never make a functional difference unless an extension requests to acquire that invalid version and somehow it manages to get through version resolution, in which case we would just return the path to the existing dotnet.exe.